src/scripts/main.js

```diff
@@ -1 +1,57 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(() => reject(new Error(`First promise was rejected`)), 3000);
+
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+});
+
+const success = (message) => {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList.add('success');
+  div.textContent = message;
+  document.body.append(div);
+};
+
+const error = (message) => {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList.add('error');
+  div.textContent = message;
+  document.body.append(div);
+};
+
+firstPromise.then(success).catch(error);
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+});
+
+secondPromise.then(success).catch(error);
+
+const thirdPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    document.addEventListener('contextmenu', () => {
+      resolve('Third promise was resolved');
+    });
+  });
+
+  document.addEventListener('contextmenu', () => {
+    document.addEventListener('click', () => {
+      resolve('Third promise was resolved');
+    });
+  });
+});
```

Comment:
The `thirdPromise` is set up to resolve only when a 'click' is followed by a 'contextmenu' or vice versa. This logic might be confusing and difficult for users to trigger intentionally. Ensure this behavior is what you intend, as it requires two specific sequential actions to resolve the promise.